openedx/features/content_tagging/models/system_defined.py

```diff
@@ -0,0 +1,80 @@
+"""
+System defined models
+"""
+from typing import Type
+
+from openedx_tagging.core.tagging.models import (
+    ModelSystemDefinedTaxonomy,
+    ModelObjectTag,
+    UserSystemDefinedTaxonomy,
+    LanguageTaxonomy,
+)
+
+from organizations.models import Organization
+from .base import ContentTaxonomyMixin
+
+
+class OrganizationModelObjectTag(ModelObjectTag):
+    """
+    ObjectTags for the OrganizationSystemDefinedTaxonomy.
+    """
+
+    class Meta:
+        proxy = True
+
+    @property
+    def tag_class_model(self) -> Type:
+        """
+        Associate the organization model
+        """
+        return Organization
+
+    @property
+    def tag_class_value(self) -> str:
+        """
+        Returns the organization name to use it on Tag.value when creating Tags for this taxonomy.
+        """
+        return "name"
+
+
+class OrganizationSystemDefinedTaxonomy(ModelSystemDefinedTaxonomy):
+    """
+    Organization based system taxonomy class.
+    """
+
+    class Meta:
+        proxy = True
+
+    @property
+    def object_tag_class(self) -> Type:
+        """
+        Returns OrganizationModelObjectTag as ObjectTag subclass associated with this taxonomy.
+        """
+        return OrganizationModelObjectTag
+
+
+class ContentLanguageTaxonomy(ContentTaxonomyMixin, LanguageTaxonomy):
+    """
+    Language system-defined taxonomy that accepts ContentTags
+    """
+
+    class Meta:
+        proxy = True
+
+
+class ContentAuthorTaxonomy(ContentTaxonomyMixin, UserSystemDefinedTaxonomy):
+    """
+    Author system-defined taxonomy that accepts Content Tags
+    """
+
+    class Meta:
+        proxy = True
+
+
+class ContentOrganizationTaxonomy(ContentTaxonomyMixin, OrganizationSystemDefinedTaxonomy):
```

Comment:
Can we just add `ContentTaxonomyMixin` to the base `OrganizationSystemDefinedTaxonomy` class? I don't see any need for two different `OrganizationSystemDefinedTaxonomy` classes.

Also: I still think we should get rid of ModelObjectTag and then we won't need any special subclass at all for OrganizationSystemDefinedTaxononomy

Reply:
Updated [here](https://github.com/openedx/edx-platform/pull/32869/commits/8f66534dea0616c0f24150cadee4a22da4a80d0b#diff-1f01045e9b5f4e8b46c76dec746b3a595ff3b11513c3a538175c406ec98719c0R40):